js/core.js

```diff
@@ -4781,9 +4788,37 @@
                     inputDisplayMode = displayAnnot.selector_ux_mode;
                 }
 
+                /**
+                 * bulkForeignKeyCreate column is set based on the following rules:
+                 *   1. defined on display property in visible-columns
+                 *   2. foreign key annotation
+                 *   3. table-display annotation
+                 *   4. display annotation on table, schema, then catalog
+                 *   5. default heuristics used when computeBulkCreateForeignKeyObject is called
+                 */
+                // check display annotation on table/schema/catalog
+                // if true, don't set anything and let the heuristics be used
+                // if false, set to false to turn off heuristics
+                if (this.table._shouldUseBulkCreateForeignKey === false) bulkCreateConstraintName = false;
+
+                if (this.table.annotations.contains(module._annotations.TABLE_DISPLAY)) {
+                    var tableAnnotation = module._getRecursiveAnnotationValue(context, this.table.annotations.get(module._annotations.TABLE_DISPLAY).content);
+
+                    if (tableAnnotation.bulk_create_foreign_key_candidates !== null) {
+                        bulkCreateConstraintName = tableAnnotation.bulk_create_foreign_key_candidates;
+                    }
+                }
+
+                // check foreign key annotation
+                // make sure `false` is not ignored since that turns this feature off
+                if (displayAnnot.bulk_create_foreign_key !== undefined && displayAnnot.bulk_create_foreign_key !== null) {
```

Comment:
Same comment about checking for the values that we allow instead of the way you currently wrote this.